### PR TITLE
Integrate ARM Docker builds to make multi-arch images

### LINKED ISSUE #3783: Integrate ARM Docker builds into Toil CI/release build process to make multi-arch tags
VG does it like this, using `docker buildx --platform`: [https://github.com/vgteam/vg/blob/f0392bec46de87b1279e3df5a5e75c0c3089cbdc/.gitlab-ci.yml#L132-L133](https://github.com/vgteam/vg/blob/f0392bec46de87b1279e3df5a5e75c0c3089cbdc/.gitlab-ci.yml#L132-L133|smart-link) . VG is using Kubernetes to back `docker buildx`, while Toil is doing the build locally with I think `docker build`, but we could do some local binfmt-misc setup and use local `docker buildx`.

We want to end up with the Toil tags on Quay having both ARM and AMD images attached to them. Ideally, Toil CI will build these, and we would have documentation for how to manually build and push both ARM and AMD versions, whether the user is on an ARM or an AMD machine (like an M1 mac).

To control the manual builds, the Toil Makefile might have to acquire some new variables for `make push_docker`, so you could say like `make push_docker platforms=linux/amd64,linux/arm64`. We also may need to add the presence of `docker buildx` or `binfmt-misc` as a dependency, at least for cross-building, and give instructions on how to install it.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1015)
┆Epic: Graviton2 Summer 2021 Project
┆Issue Number: TOIL-1015

### PR DESCRIPTION
**Pull request by jonathanxu18**
Attempts to fix #3783.

Implements `docker buildx` to make multi-arch images and includes some adjustments for ARM in the Dockerfile.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * PR submitter writes their recommendation for a changelog entry here

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Comment by adamnovak** [comment]
Looks like the Docker builds are definitely running on the buildkit pods on Kubernetes now, but they're still bailing out on the Gitlab runner for no apparent reason and claiming to succeed.

Here's the buildkit builder log; you can see it complains about a session finishing and a context canceling.

```
time="2021-10-19T17:42:03Z" level=debug msg="> creating swzbo8ncreryrfpbnbhwc7cws [/bin/sh -c mkdir -p $(go env GOPATH)/src/github.com/sylabs &&         cd $(go env GOPATH)/src/github.com/sylabs &&         git clone https://github.com/sylabs/singularity.git &&         cd singularity &&         git checkout v3.4.2 &&         ./mconfig &&         cd ./builddir &&         make -j4 &&         make install &&         cd &&         rm -Rf $(go env GOPATH)         && mkdir -p /usr/local/libexec/toil &&         mv /usr/local/bin/singularity /usr/local/libexec/toil/singularity-real]" span="[linux/arm64  7/26] RUN mkdir -p $(go env GOPATH)/src/github.com/sylabs &&         cd $(go env GOPATH)/src/github.com/sylabs &&         git clone https://github.com/sylabs/singularity.git &&         cd singularity &&         git checkout v3.4.2 &&         ./mconfig &&         cd ./builddir &&         make -j4 &&         make install &&         cd &&         rm -Rf $(go env GOPATH)         && mkdir -p /usr/local/libexec/toil &&         mv /usr/local/bin/singularity /usr/local/libexec/toil/singularity-real"
time="2021-10-19T17:46:48Z" level=debug msg="> creating xyk8815qbaeufujc92x33bk67 [/bin/sh -c rm toil-5.6.0a1.tar.gz]" span="[linux/amd64 25/26] RUN rm toil-5.6.0a1.tar.gz"
time="2021-10-19T17:46:49Z" level=debug msg="> creating 73tfgxew0vf5dfmbjdtdvcdq2 [/bin/sh -c echo '[ ! -z \"$TERM\" -a -r /etc/motd ] && cat /etc/motd' >> /etc/bash.bashrc         && printf '\\nThis is the Toil appliance. You can run your Toil script directly on the appliance.\\nRun toil <workflow>.py --help to see all options for running your workflow.\\nFor more information see http://toil.readthedocs.io/en/latest/\\n\\nCopyright (C) 2015-2020 Regents of the University of California\\n\\nVersion: quay.io/ucsc_cgl/toil:5.6.0a1-8bfd97c3963b020f36d5b8afbb7b09f140da7332-py3.8\\n\\n' > /etc/motd]" span="[linux/amd64 26/26] RUN echo '[ ! -z \"$TERM\" -a -r /etc/motd ] && cat /etc/motd' >> /etc/bash.bashrc         && printf '\\nThis is the Toil appliance. You can run your Toil script directly on the appliance.\\nRun toil <workflow>.py --help to see all options for running your workflow.\\nFor more information see http://toil.readthedocs.io/en/latest/\\n\\nCopyright (C) 2015-2020 Regents of the University of California\\n\\nVersion: quay.io/ucsc_cgl/toil:5.6.0a1-8bfd97c3963b020f36d5b8afbb7b09f140da7332-py3.8\\n\\n' > /etc/motd"
time="2021-10-19T17:47:53Z" level=debug msg="> creating n7ugeu6te0vjg5lijo4cvqo4l [/bin/sh -c rm toil-5.6.0a1.tar.gz]" span="[linux/amd64 25/26] RUN rm toil-5.6.0a1.tar.gz"
time="2021-10-19T17:47:54Z" level=debug msg="> creating 0zwk5xhvgpynng93fkv4gta04 [/bin/sh -c echo '[ ! -z \"$TERM\" -a -r /etc/motd ] && cat /etc/motd' >> /etc/bash.bashrc         && printf '\\nThis is the Toil appliance. You can run your Toil script directly on the appliance.\\nRun toil <workflow>.py --help to see all options for running your workflow.\\nFor more information see http://toil.readthedocs.io/en/latest/\\n\\nCopyright (C) 2015-2020 Regents of the University of California\\n\\nVersion: quay.io/ucsc_cgl/toil:5.6.0a1-8bfd97c3963b020f36d5b8afbb7b09f140da7332-py3.6\\n\\n' > /etc/motd]" span="[linux/amd64 26/26] RUN echo '[ ! -z \"$TERM\" -a -r /etc/motd ] && cat /etc/motd' >> /etc/bash.bashrc         && printf '\\nThis is the Toil appliance. You can run your Toil script directly on the appliance.\\nRun toil <workflow>.py --help to see all options for running your workflow.\\nFor more information see http://toil.readthedocs.io/en/latest/\\n\\nCopyright (C) 2015-2020 Regents of the University of California\\n\\nVersion: quay.io/ucsc_cgl/toil:5.6.0a1-8bfd97c3963b020f36d5b8afbb7b09f140da7332-py3.6\\n\\n' > /etc/motd"
time="2021-10-19T17:49:41Z" level=debug msg="> creating u7jxz3u2rvdfotxkjc07qvrip [/bin/sh -c rm toil-5.6.0a1.tar.gz]" span="[linux/amd64 25/26] RUN rm toil-5.6.0a1.tar.gz"
time="2021-10-19T17:49:42Z" level=debug msg="> creating u7cjd74x369rjqykweppxq383 [/bin/sh -c echo '[ ! -z \"$TERM\" -a -r /etc/motd ] && cat /etc/motd' >> /etc/bash.bashrc         && printf '\\nThis is the Toil appliance. You can run your Toil script directly on the appliance.\\nRun toil <workflow>.py --help to see all options for running your workflow.\\nFor more information see http://toil.readthedocs.io/en/latest/\\n\\nCopyright (C) 2015-2020 Regents of the University of California\\n\\nVersion: quay.io/ucsc_cgl/toil:5.6.0a1-8bfd97c3963b020f36d5b8afbb7b09f140da7332-py3.7\\n\\n' > /etc/motd]" span="[linux/amd64 26/26] RUN echo '[ ! -z \"$TERM\" -a -r /etc/motd ] && cat /etc/motd' >> /etc/bash.bashrc         && printf '\\nThis is the Toil appliance. You can run your Toil script directly on the appliance.\\nRun toil <workflow>.py --help to see all options for running your workflow.\\nFor more information see http://toil.readthedocs.io/en/latest/\\n\\nCopyright (C) 2015-2020 Regents of the University of California\\n\\nVersion: quay.io/ucsc_cgl/toil:5.6.0a1-8bfd97c3963b020f36d5b8afbb7b09f140da7332-py3.7\\n\\n' > /etc/motd"
time="2021-10-19T21:58:22Z" level=debug msg="session finished: <nil>" spanID=e74f3222cac443e7 traceID=7d589db7adb2a09c42e59a9b43dafd09
time="2021-10-19T21:58:22Z" level=error msg="/moby.buildkit.v1.Control/Solve returned error: rpc error: code = Canceled desc = context canceled\n"
time="2021-10-19T21:58:23Z" level=debug msg="remove snapshot" key=siwb5ive2khjpbbm1w7jts5v4 snapshotter=overlayfs
time="2021-10-19T21:58:23Z" level=debug msg="remove content" key="sha256:033c410d53efaab9e847168b94fea14d551788200a15f4d2424c84e4cd2ab8a2"
time="2021-10-19T21:58:23Z" level=debug msg="remove content" key="sha256:4a6759446930ee402a2091605c0911f237ad18b5c250a535d11563d9402eb15a"
time="2021-10-19T21:58:23Z" level=debug msg="remove content" key="sha256:ac20e7c3411697a4a85c1cb30850f77b8238e0a0ef0e176ca8f0e763066eed4b"
time="2021-10-19T21:58:23Z" level=debug msg="remove content" key="sha256:b722e2654241f9681f4719dce7aa16a2f0c35769e17a636f5b39a33967d1aeb8"
time="2021-10-19T21:58:23Z" level=debug msg="schedule snapshotter cleanup" snapshotter=overlayfs
time="2021-10-19T21:58:23Z" level=debug msg="schedule content cleanup"
time="2021-10-19T21:58:23Z" level=debug msg="removed content" digest="sha256:033c410d53efaab9e847168b94fea14d551788200a15f4d2424c84e4cd2ab8a2"
time="2021-10-19T21:58:23Z" level=debug msg="removed content" digest="sha256:4a6759446930ee402a2091605c0911f237ad18b5c250a535d11563d9402eb15a"
time="2021-10-19T21:58:23Z" level=debug msg="removed content" digest="sha256:ac20e7c3411697a4a85c1cb30850f77b8238e0a0ef0e176ca8f0e763066eed4b"
time="2021-10-19T21:58:23Z" level=debug msg="removed content" digest="sha256:b722e2654241f9681f4719dce7aa16a2f0c35769e17a636f5b39a33967d1aeb8"
time="2021-10-19T21:58:23Z" level=debug msg="content garbage collected" d="448.404µs"
time="2021-10-19T21:58:23Z" level=debug msg="removed snapshot" key=buildkit/53/siwb5ive2khjpbbm1w7jts5v4 snapshotter=overlayfs
time="2021-10-19T21:58:23Z" level=debug msg="snapshot garbage collected" d=249.246525ms snapshotter=overlayfs
```

On the Gitlab side, it looks like both https://ucsc-ci.com/databiosphere/toil/-/jobs/100210/raw and https://ucsc-ci.com/databiosphere/toil/-/jobs/100212/raw stop at about the same place, in the ARM Singularity build, near a complaint like:

```
#38 1220.8 time="2021-10-19T17:58:57Z" level=error msg="failure getting variant" error="getCPUInfo for pattern: Cpu architecture: not found"
```

I'm not sure why that's happening; it looks like the ARM Singularity build configures itself for ARM properly:

```
#38 415.5 => building Makefile ...
#38 421.7 => generating singularity.spec ...
#38 423.3 => project singularity setup with :
#38 423.3     - host arch: aarch64
#38 423.3     - host wordsize: 64-bit
#38 423.3     - host C compiler: cc
#38 423.3     - host Go compiler: /usr/bin/go
#38 423.3     - host system: unix
#38 423.3       ---
#38 423.3     - target arch: aarch64
#38 423.3     - target wordsize: 64-bit
#38 423.3     - target C compiler: cc
#38 423.3       ---
#38 423.3     - config profile: release
#38 423.3       ---
#38 423.3     - SUID install: yes
#38 423.3     - Network plugins: yes
#38 423.3       ---
#38 423.3     - verbose: no
#38 423.3       ---
#38 423.3     - cryptsetup: /sbin/cryptsetup
#38 423.3       ---
#38 423.3     - version: 3.4.2
```

I've seen some complaints that this happens in builds that themselves use containers or consult `/proc/cpuinfo`, but I've yet to pin down exactly what program is making this complaint or what it thinks it is doing. 

@jonathanxu18 Have you gotten the ARM Docker build to succeed past the Singularity build step on an x86 host with emulation set up, when you run it manually?

Maybe we should look into not actually doing the Singularity build every time we make the Docker images, if we're going to emulate the whole thing. Ubuntu still doesn't ship Singularity 3, but Debian does, so we could try installing Debian's debs onto our Ubuntu base, or swap to a Debian base.

On Ubuntu 20.04 I managed to install https://debian.osuosl.org/debian/pool/main/s/singularity-container/singularity-container_3.5.2+ds1-1_amd64.deb after installing its `containernetworking-plugins` dependency. We probably could always grab an up-to-date deb for a given architecture with:

```
wget "https://debian.osuosl.org/debian/pool/main/s/singularity-container/$(curl -sSL 'https://debian.osuosl.org/debian/pool/main/s/singularity-container/' | grep -o 'singularity-container_3[^"]*amd64.deb' | head -n1)"
```

Or similarly for `arm64`. Then we just `apt install ca-certificates containernetworking-plugins libarchive13 libc6 libseccomp2 python squashfs-tools e2fsprogs` to make sure we have the dependencies, and `dpkg -i singularity-container_3*.deb`, and we should be able to skip the whole Singularity build.

**Review by adamnovak — COMMENTED** [review]
This looks like it ought to work! But we should inspect what gets pushed up to Quay and see if arm64 and x86_64 versions are actually pullable in the tags this makes for CI, before we merge it.

**Comment by adamnovak** [comment]
Looks like the ARM docker builds actually work! You can see on Quay that the tags this PR build have arm64 versions, because they have 2 penguins in their entries in the table of tags:

image

Unfortunately, the CWL Kubernetes tests all failed with:

Log from job "'CWLJob' cat e60329bd-1e50-4851-bb19-269b8e581909 v6" follows:
=========>
    [91mERROR  : Failed to create container namespaces
    [0mTraceback (most recent call last):
      File "/usr/local/lib/python3.8/dist-packages/toil/worker.py", line 402, in workerScript
        job._runner(jobGraph=None, jobStore=jobStore, fileStore=fileStore, defer=defer)
      File "/usr/local/lib/python3.8/dist-packages/toil/job.py", line 2362, in _runner
        returnValues = self._run(jobGraph=None, fileStore=fileStore)
      File "/usr/local/lib/python3.8/dist-packages/toil/job.py", line 2283, in _run
        return self.run(fileStore)
      File "/usr/local/lib/python3.8/dist-packages/toil/cwl/cwltoil.py", line 2011, in run
        raise cwltool.errors.WorkflowException(status)
    cwltool.errors.WorkflowException: permanentFail
    [2021-11-04T23:31:50+0000] [MainThread] [E] [toil.worker] Exiting the worker because of a failed job on host toiltest-toil-980340fd-b174-40d5-bb30-b34efe329825-1-zvggs
<=========

I have no idea why they did that. It's not that the jobs didn't schedule on Kubernetes; maybe we broke something about Singularity, and it doesn't actually work for CWL as we ship it in the new appliance? Maybe we really do need to upgrade the base Ubuntu?

**Comment by adamnovak** [comment]
For debugging this, I've figured out that even Singularity with `-u` won't work in a normal Docker container, by looking at https://github.com/hpcng/singularity/issues/5857. Docker adds some extra security-confining stuff that we'd have to turn off, and I'm not sure how to do that. We *probably* can test locally with `--privileged` in the Docker call, or we can test on Kubernetes, where Singularity can run in an unprivileged container. I've been doing something like this:

```
kubectl delete pod adamnovak-pod    

kubectl apply -f - <<'EOF'
apiVersion: v1
kind: Pod
metadata:
  name: adamnovak-pod
spec:
  containers:
  - name: anovak-container
    image: quay.io/ucsc_cgl/toil:5.6.0a1-52c13224f47dbd94a3bcdb4caaba086a7a683b69-py3.6
    imagePullPolicy: Always
    env:
    args: ["sleep", "infinity"]
    resources:
      requests:
        cpu: 4000m
        memory: "4Gi"
        ephemeral-storage: "10Gi"
      limits:
        cpu: 4000m
        memory: "4Gi"
        ephemeral-storage: "10Gi"
    volumeMounts:
      - mountPath: /scratch
        name: scratch
  restartPolicy: Never
  volumes:
  - name: scratch
    emptyDir: {}
EOF

kubectl exec -ti adamnovak-pod -- /bin/bash
```

**Comment by adamnovak** [comment]
In the Kubernetes pod, I've been running:

```
cd ~
mkdir test
cd ~/test
singularity build -s -F ./sandbox docker://ubuntu:20.04
'singularity' 'exec' '-w' '-u' '-B' '' '--pwd' '/mnt' './sandbox' ls -lah /
```

When the pod is `quay.io/ucsc_cgl/toil:5.5.0-py3.6`, it works and shows me the root of a Singularity container with Singularity-related symlinks like:

```
lrwxrwxrwx    1 root root   27 Nov  8 16:09 .exec -> .singularity.d/actions/exec
```

But with quay.io/ucsc_cgl/toil:5.6.0a1-52c13224f47dbd94a3bcdb4caaba086a7a683b69-py3.6 from this PR it fails:

```
FATAL:   container creation failed: mount /etc/localtime->/etc/localtime error: while mounting /etc/localtime: mount source /etc/localtime doesn't exist
```

We did jump a few Singularity versions, so maybe we need to so something to keep the new Singularity happy.

**Comment by adamnovak** [comment]
I blame this part of Debian's `/etc/singularity/singularity.conf` where the default is overridden and we say that all containers should get `/etc/hosts` and `/etc/localtime`. I think the Kubernetes pods are missing `/etc/localtime` (which configures the timezone), and the Debian Singularity packagers have "improved" it to expose the configured timezone to containers, but it isn't working because the file isn't there to mount.

```
# BIND PATH: [STRING]
# DEFAULT: Undefined
# Define a list of files/directories that should be made available from within
# the container. The file or directory must exist within the container on
# which to attach to. you can specify a different source and destination
# path (respectively) with a colon; otherwise source and dest are the same.
# NOTE: these are ignored if singularity is invoked with --contain.
#bind path = /etc/singularity/default-nsswitch.conf:/etc/nsswitch.conf
#bind path = /opt
#bind path = /scratch
bind path = /etc/localtime
bind path = /etc/hosts
```

**Comment by adamnovak** [comment]
We're getting a timeout in just one of the tests for the Kubernetes CWL conformance tests now:

```
Test 170 timed out: toil-cwl-runner --disableCaching=True --clean=always --logDebug --setEnv=SINGULARITY_DOCKER_HUB_MIRROR --batchSystem=kubernetes --outdir=/tmp/tmpp6mozk32 --quiet tests/step-valuefrom4-wf.cwl tests/empty.json
```

We need to see if we can reproduce that issue outside of CI, from a leader with access to GI Kubernetes, and if so if we can fix it. I don't see any reasonable way it would have anything to do with this PR's changes; it's just a single test.

**Review by adamnovak — CHANGES_REQUESTED** [review]
Actually, we need to keep the part where we move Singularity's binary (https://github.com/DataBiosphere/toil/blob/312b6e1f221ee7f7f187dd6dbfce1aecffd00e09/docker/Dockerfile.py#L114) and the whole block where we add our scripts and the Singularity wrapper (https://github.com/DataBiosphere/toil/blob/312b6e1f221ee7f7f187dd6dbfce1aecffd00e09/docker/Dockerfile.py#L118-L124). I missed that we were dropping those and we don't want to be.

File: docker/Dockerfile.py [hunk 1]
```diff
@@ -35,11 +35,14 @@
                          'wget',
                          'curl',
                          'openssh-server',
-                         'mesos=1.0.1-2.0.94.ubuntu1604',
```
Comment (DailyDreaming) [inline docker/Dockerfile.py:38]: Don't we still need mesos?

File: Makefile [hunk 2]
```diff
@@ -182,7 +181,7 @@ clean_docker:
 
 push_docker: docker
 	# Weird if logic is so we fail if all the pushes fail
-	for i in $$(seq 1 6); do if [[ $$i == "6" ]] ; then exit 1 ; fi ; docker push $(docker_image):$(TOIL_DOCKER_TAG) && break || sleep 60; done
+	cd docker ; for i in $$(seq 1 6); do if [[ $$i == "6" ]] ; then exit 1 ; fi ; docker buildx build --platform linux/amd64,linux/arm64 --push --tag=$(docker_image):$(TOIL_DOCKER_TAG) -f Dockerfile . && break || sleep 60; done
 	for i in $$(seq 1 6); do if [[ $$i == "6" ]] ; then exit 1 ; fi ; docker push $(grafana_image):$(TOIL_DOCKER_TAG) && break || sleep 60; done
 	for i in $$(seq 1 6); do if [[ $$i == "6" ]] ; then exit 1 ; fi ; docker push $(prometheus_image):$(TOIL_DOCKER_TAG) && break || sleep 60; done
 	for i in $$(seq 1 6); do if [[ $$i == "6" ]] ; then exit 1 ; fi ; docker push $(mtail_image):$(TOIL_DOCKER_TAG) && break || sleep 60; done
```
Comment (DailyDreaming) [inline Makefile:187]: Should we be building the metrics containers as ARM as well?

**Review by DailyDreaming — CHANGES_REQUESTED** [review]
Is it possible to have a test that runs on ARM?  See comments below.

File: docker/Dockerfile.py (hunk 1 above)
Comment (adamnovak) [reply docker/Dockerfile.py:38]: We do, but not in ARM, where it's not available. It gets installed on x86 later on.

File: Makefile (hunk 2 above)
Comment (adamnovak) [reply Makefile:187]: We currently only use them on Mesos, and we have no Mesos vuild for ARM and therefore no ARM Mesos clusters.

Maybe if we decide to build out the dashboard on Kubernetes we'd need ARM versions of these images. Even then, the Right Way to do the dashboard on Kubernetes would probably be to mount all the Toil-specific config files and do all the Toil-specific customization at the Kubernetes level, on top of stock grafana, mtail, and prometheus images.

**Comment by adamnovak** [comment]
> Is it possible to have a test that runs on ARM? See comments below.

Getting ARM actually under test is going to require building something to launch an ARM instance from the tests (or some hackery to set up binfmt-misc and somehow use it to run the "wrong" arch of a multi-arch Docker image).

We're eventually going to need ARM support for `toil launch-cluster`, which will use these images and be feasible to test, but that's not part of #3783.

There is also #3890 which is about getting CWL on ARM under test, but that's its own issue.

I suppose we could add some kind of test to see if the multi-arch images actually are uploaded? But I don't think that's machinery we're going to keep long term, because we're going to actually use the ARM images in the next step of the project.

**Review by adamnovak — APPROVED** [review]
I think this is working properly!

File: Makefile (hunk 2 above)
Comment (DailyDreaming) [reply Makefile:189]: I'll make an issue based on this then.

**Review by DailyDreaming — APPROVED** [review]
@adamnovak Your points make sense and this looks good to me now too.

**Pull request merged**